soapui/src/main/java/com/eviware/soapui/support/swing/FunctionalMenu.java

```diff
@@ -0,0 +1,139 @@
+package com.eviware.soapui.support.swing;
+
+import com.eviware.soapui.SoapUI;
+import com.eviware.soapui.impl.wsdl.WsdlProject;
+import com.eviware.soapui.impl.wsdl.WsdlTestSuite;
+import com.eviware.soapui.impl.wsdl.actions.support.ShowDesktopPanelAction;
+import com.eviware.soapui.impl.wsdl.actions.teststep.ToggleDisableTestStepAction;
+import com.eviware.soapui.impl.wsdl.testcase.WsdlTestCase;
+import com.eviware.soapui.impl.wsdl.teststeps.WsdlTestStep;
+import com.eviware.soapui.model.ModelItem;
+import com.eviware.soapui.model.support.ModelSupport;
+import com.eviware.soapui.model.tree.SoapUITreeNode;
+import com.eviware.soapui.model.workspace.Workspace;
+import com.eviware.soapui.support.action.SoapUIActionMapping;
+import com.eviware.soapui.support.action.support.DefaultActionMapping;
+import com.eviware.soapui.support.action.swing.ActionList;
+import com.eviware.soapui.support.action.swing.ActionListBuilder;
+import com.eviware.soapui.support.action.swing.ActionSupport;
+import com.eviware.soapui.support.action.swing.SwingActionDelegate;
+import javax.swing.*;
+import java.awt.*;
+
+
+public class FunctionalMenu {
+    public static JMenu getMenu(String name){
+        for (int i = 0; i<SoapUI.getMenuBar().getMenuCount(); i++){
+            if (SoapUI.getMenuBar().getMenu(i).getText().equals(name)){
+                return SoapUI.getMenuBar().getMenu(i);
+            }
+        }
+        return null;
+    }
```

Comment:
this function uses SoapUI.getMenuBar() at 3 places. One particular variable should be extracted before "for" loop.